### PR TITLE
Stack Monitoring parity test for Logstash

### PR DESCRIPTION
**Pull request by ycombinator**
This PR introduces Stack Monitoring parity tests for Logstash. Specifically, it introduces a playbook that will install Logstash and setup internal collection for it. Then it will monitor the same Logtash node with external (Metricbeat) collection. Finally, it will compare the internally-and externally-collected monitoring documents for structural parity.

### Testing this PR

1. Visit https://internal-ci.elastic.co/view/Release/job/elastic+release-manager+master+unified-snapshot/lastSuccessfulBuild/console, and note the URL for the latest successful snapshot build. The URL will look like `https://snapshots.elastic.co/MAJOR.MINOR.PATH-SHA`.

2. Tell the parity tests to use the latest release snapshot. Edit `playbooks/monitoring/buildenv.sh` and set `ES_BUILD_URL` to the URL you noted in step 1. For example:
   ```
   export ES_BUILD_URL=https://snapshots.elastic.co/8.0.0-ae5632f0
   ```

3. Run the monitoring parity tests for Logstash.
   ```
   AIT_STACK_PRODUCT=logstash ./playbooks/monitoring/buildenv.sh
   ```

4. Verify that the tests pass.

### CODE REVIEW

**Comment by ycombinator** [comment]
This PR is now at a point where the Logstash parity testing playbook itself is in good shape. The failures I'm seeing by running this playbook now are strictly due to parity comparisons. So from this point forward I'll be posting commits to this PR or links to other PRs that will help achieve parity.

**Comment by ycombinator** [comment]
Blocked on https://github.com/elastic/beats/pull/12881.

**Comment by ycombinator** [comment]
Blocked on https://github.com/elastic/beats/pull/12883.

**Comment by ycombinator** [comment]
Blocked on https://github.com/elastic/logstash/issues/10963.

**Comment by ycombinator** [comment]
Blocked on https://github.com/elastic/beats/pull/12917.

**Comment by ycombinator** [comment]
Blocked on https://github.com/elastic/beats/pull/12929.

**Comment by ycombinator** [comment]
Blocked on https://github.com/elastic/beats/pull/12989.

**Comment by ycombinator** [comment]
Blocked on ~https://github.com/elastic/beats/pull/13082~.

**Comment by ycombinator** [comment]
**UPDATE**: At this point I've been able to achieve parity between `type:logstash_stats` documents. Now the parity tests are failing on differences between `type:logstash_state` documents. Continuing to investigate and make them pass...

**Comment by ycombinator** [comment]
Blocked on https://github.com/elastic/beats/pull/13132.

**Comment by ycombinator** [comment]
Blocked on https://github.com/elastic/beats/pull/13133.

File: playbooks/monitoring/logstash/docs_compare.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    doc_types = []
+    for doc_type_filename in (f for f in files if os.path.isfile(os.path.join(docs_path, f))):
+        name, ext = doc_type_filename.split('.')
+        doc_types.append(name)
+    return doc_types
+
+def get_doc(docs_path, doc_type):
+    with open(os.path.join(docs_path, doc_type + ".json")) as f:
+        data = f.read()
+    f.closed
```
Comment (cachedout) [inline playbooks/monitoring/logstash/docs_compare.py]: I don't understand this line. The `with` [should be closing the file](https://docs.python.org/3/tutorial/inputoutput.html?highlight=closed#reading-and-writing-files) and this just looks like it's a no-op. Could you please clarify what the intention was here?

Comment (ycombinator) [reply playbooks/monitoring/logstash/docs_compare.py]: You're right, it's a no-op and totally unnecessary. Good catch! I'll remove it from this PR and make a separate PR to remove it from other products' parity test code as well. Thanks!

Comment (ycombinator) [reply playbooks/monitoring/logstash/docs_compare.py]: Fixed in 61f2e959c4167c14c0b0ae2218a89b6ca94a82e3.

Comment (ycombinator) [reply playbooks/monitoring/logstash/docs_compare.py]: Separate PR to remove the line from other products' parity comparison scripts filed here: https://github.com/elastic/elastic-stack-testing/pull/295

**Comment by ycombinator** [comment]
@cachedout I've addressed your feedback. This PR is ready for your 👀 again. Thanks!

File: playbooks/monitoring/logstash/docs_compare.py (hunk 1 above)
Comment (cachedout) [reply playbooks/monitoring/logstash/docs_compare.py]: Great! Thank you!

**Pull request merged**